ext/date/date_core.c

```diff
@@ -3527,19 +3529,46 @@ date_initialize(int argc, VALUE *argv, VALUE self)
 
 /*
  * call-seq:
- *    Date.commercial([cwyear=-4712[, cweek=1[, cwday=1[, start=Date::ITALY]]]])  ->  date
+ *   Date.commercial(cwyear=-4712, cweek=1, cwday=1, start=Date::ITALY) -> date
+ *
+ * Returns a new \Date object constructed from the arguments.
+ *
+ * Argument +cwyear+ gives the year, and should be an integer.
+ *
+ * Argument +cweek+ gives the index of the week within the year,
+ * and should be in range (1..53) or (-53..-1);
+ * in some years, 53 or -53 will be out-of-range;
+ * if negative, counts backward from the end of the year:
+ *
+ *   Date.commercial(2022, 1, 1).to_s  # => "2022-01-03"
+ *   Date.commercial(2022, 52, 1).to_s # => "2022-12-26"
+ *
+ * Argument +cwday+ gives the indes of the weekday within the week,
+ * and should be in range (1..7) or (-7..-1);
+ * 1 or -7 is Monday;
+ * if negative, counts backward from the end of the week:
+ *
+ *   Date.commercial(2022, 1, 1).to_s  # => "2022-01-03"
+ *   Date.commercial(2022, 1, -7).to_s # => "2022-01-03"
  *
- * Creates a date object denoting the given week date.
+ * When +cweek+ is 1:
  *
- * The week and the day of week should be a negative or a positive
- * number (as a relative week/day from the end of year/week when
- * negative).  They should not be zero.
+ * - If January 1 is a Friday, Saturday, Sunday, or Monday,
+ *   the first week begins in the year +cwyear+:
  *
- *    Date.commercial(2001)	#=> #<Date: 2001-01-01 ...>
- *    Date.commercial(2002)	#=> #<Date: 2001-12-31 ...>
- *    Date.commercial(2001,5,6)	#=> #<Date: 2001-02-03 ...>
+ *     # January 1 is a Monday.
+ *     Date.commercial(2001, 1, 1).to_s # => "2001-01-01"
+ *     Date.commercial(2001, 1, 7).to_s # => "2001-01-07"
  *
- * See also ::jd and ::new.
+ * - Otherwise, the first begins in the year <tt>cwyear + 1</tt>:
+ *
+ *     # January 1 is a Thursday.
+ *     Date.commercial(2004, 1, 1).to_s # => "2003-12-29"
+ *     Date.commercial(2004, 1, 7).to_s # => "2004-01-04"
```

Comment:
1. Monday is included with FSS, I'm pretty sure.  Add it?
2. Suggested wording for the 'otherwise' clause?

Reply:
I would word it something like this:

```
- If January 1 is a Friday, Saturday, Sunday,
  the first week begins in the week after:
- Otherwise, the first week is the week of
  January 1, which may mean some of the
  days fall on the year before:
```

Reply:
Done.